enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/discovery/Cluster.java

```diff
@@ -357,26 +357,46 @@ public CoreClusterMember getDbWithRole( Role role )
         return getDbWithAnyRole( role );
     }
 
+    public List<CoreClusterMember> getAllDbsWithRole( Role role )
+    {
+        return getAllDbsWithAnyRole( role );
+    }
+
     public CoreClusterMember getDbWithRole( String dbName, Role role )
     {
         return getDbWithAnyRole( dbName, role );
     }
 
+    public List<CoreClusterMember> getAllDbsWithRole( String dbName, Role role )
+    {
+        return getAllDbsWithAnyRole( dbName, role );
+    }
+
     public CoreClusterMember getDbWithAnyRole( Role... roles )
     {
         String dbName = CausalClusteringSettings.database.getDefaultValue();
         return getDbWithAnyRole( dbName, roles );
     }
 
+    public List<CoreClusterMember> getAllDbsWithAnyRole( Role... roles )
+    {
+        String dbName = CausalClusteringSettings.database.getDefaultValue();
+        return getAllDbsWithAnyRole( dbName, roles );
+    }
+
     public CoreClusterMember getDbWithAnyRole( String dbName, Role... roles )
+    {
+        return getAllDbsWithAnyRole( dbName, roles ).stream().findFirst().orElse( null );
+    }
+
+    public List<CoreClusterMember> getAllDbsWithAnyRole( String dbName, Role... roles )
```

Comment:
Generally naming around these methods is a bit confusing. Get all databases for database. Not saying we should refactor all the things though. Maybe get all cluster members?